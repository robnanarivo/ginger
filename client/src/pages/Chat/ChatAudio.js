import React from 'react';
import { Grid, Typography } from '@mui/material';
import ReactAudioPlayer from 'react-audio-player';

const ChatAudio = ({
  sender,
  url,
  timestamp,
  isSelf,
}) => {
  let align = 'left';
  if (isSelf) {
    align = 'right';
  }
  return (
    <Grid container sx={{ p: 2 }}>
      <Grid item xs={12} align={align}>
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{sender}</Typography>
        <ReactAudioPlayer
          src={url}
          controls
        />
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{timestamp}</Typography>
      </Grid>
    </Grid>
  );
};

export default ChatAudio;
