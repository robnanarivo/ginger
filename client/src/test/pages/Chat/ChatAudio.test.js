import React from 'react';
import renderer from 'react-test-renderer';
import ChatAudio from '../../../pages/Chat/ChatAudio';
import 'jest-styled-components';

describe('ChatAudio unit test', () => {
  it('ChatAudio Snapshot test', () => {
    const tree = renderer
      .create(<ChatAudio sender="You" isSelf timestamp="20211202" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
