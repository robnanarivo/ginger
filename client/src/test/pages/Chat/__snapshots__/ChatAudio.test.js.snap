// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatAudio unit test ChatAudio Snapshot test 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1fkyx8n-MuiGrid-root"
>
  <div
    align="right"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      You
    </p>
    <audio
      autoPlay={false}
      className="react-audio-player "
      controls={true}
      id=""
      loop={false}
      muted={false}
      preload="metadata"
      style={Object {}}
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      20211202
    </p>
  </div>
</div>
`;
